parsers/ECO2MIX.py

```diff
@@ -0,0 +1,295 @@
+#!/usr/bin/env python3
+
+import json
+import math
+import os
+from datetime import datetime, timedelta
+from logging import Logger, getLogger
+
+import arrow
+import numpy as np
+import pandas as pd
+import requests
+from bs4 import BeautifulSoup
+
+from electricitymap.contrib.config import EXCHANGES_CONFIG, ZONES_CONFIG
+from parsers.lib.config import refetch_frequency
+from parsers.lib.exceptions import ParserException
+
+from .lib.validation import validate, validate_production_diffs
+
+# From RTE domain to EM domain
+MAP_ZONES = {
+    "France": "FR",
+    **{
+        subzone.replace("FR-", ""): subzone
+        for subzone in ZONES_CONFIG["FR"].get("subZoneNames", [])
+    },
+    "BE": "BE",
+    "CH": "CH",
+    "DE": "DE",
+    "ES": "ES",
+    "GB": "GB",
+    "IT": "IT-NO",
+    "LU": "LU",
+}
+
+MAP_MODES = {
+    "NuclÃ©aire": "production.nuclear",
+    "Thermique": "production.unknown",
+    "Eolien": "production.wind",
+    "Solaire": "production.solar",
+    "Hydraulique": "production.hydro",
+    "Bioénergies": "production.biomass",
+    "Consommation": "consumption",
+    "Autres": "production.unknown",
+    "Charbon": "production.coal",
+    "Fioul": "production.oil",
+    "Gaz": "production.gas",
+    "Pompage": "storage.hydro",
+    "Destockage": "storage.hydro",
+    "Batterie_Injection": "storage.battery",
+    "Batterie_Soutirage": "storage.battery",
+    "Stockage": "storage.battery",
+    "Solde": "IGNORED",
+    "Co2": "IGNORED",
+    "Taux de Co2": "IGNORED",
+    "PrÃ©visionJ": "IGNORED",
+    "PrÃ©visionJ-1": "IGNORED",
+}
+
+# validations for each region
+VALIDATIONS = {
+    "FR-ARA": ["unknown", "nuclear", "hydro"],
+}
+
+
+def query(url_type_arg, session=None, target_datetime=None):
```

Comment:
Could be smart to add types here (and the other functions) for better linting and DX but it's technically not required.